### PR TITLE
Validate grpc port info in GRPCServer ctor

### PR DESCRIPTION
**Pull request by ckeshava**
## High Level Overview of Change
Fix https://github.com/XRPLF/rippled/issues/4015, https://github.com/XRPLF/rippled/issues/4557: GRPCServer and ServerHandler are created during the construction of Application instance. Do not validate GRPC IP/Port/Protocol information in ServerHandler. Instead, pass the responsibility of grpc port information validation to GRPCServer constructor. This ensures that configuration file can specify the port_grpc section.

### Context of Change
Based on suggestions from @ximinez , I implemented this code change. @cjcobb23 suggests an alternative solution to the same issue https://github.com/XRPLF/rippled/issues/4015

I welcome any other suggestions too. I felt this is a clean way to handle this problem without breaking backwards compatibility.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

This pertains to renaming of variables inside the codebase. It should suffice to test it through the existing unit tests.

### CODE REVIEW

File: src/ripple/app/main/GRPCServer.cpp [hunk 1]
```diff
@@ -427,9 +428,9 @@ GRPCServerImpl::GRPCServerImpl(Application& app)
     : app_(app), journal_(app_.journal("gRPC Server"))
 {
     // if present, get endpoint from config
-    if (app_.config().exists("port_grpc"))
+    if (app_.config().exists(SECTION_PORT_GRPC))
     {
-        const auto& section = app_.config().section("port_grpc");
+        Section section = app_.config().section(SECTION_PORT_GRPC);
```
Comment (ximinez) [inline src/ripple/app/main/GRPCServer.cpp:433]: Don't make a copy of the `Section`. 
```
Section const& section = ...
```

**Review by ximinez — CHANGES_REQUESTED** [review]
This is only a partial review after taking a quick glance

File: src/ripple/app/main/GRPCServer.cpp (hunk 1 above)
Comment (ckeshava) [reply src/ripple/app/main/GRPCServer.cpp]: fixed in dc4a1beb066afbce6ed123923b1d134aea68e372

**Review by ximinez — CHANGES_REQUESTED** [review]
One minor suggestion. Also, please merge in the latest develop branch.

File: src/ripple/core/ConfigSections.h [hunk 2]
```diff
@@ -101,6 +101,8 @@ struct ConfigSection
 #define SECTION_SWEEP_INTERVAL "sweep_interval"
 #define SECTION_NETWORK_ID "network_id"
 
+#define SECTION_PORT_GRPC "port_grpc"
```
[the remark this answers is not in the document]
Comment (ckeshava) [reply src/ripple/core/ConfigSections.h]: The last 6 `#defines` are not in alphabetical order, I've fixed them too

**Review by ximinez — APPROVED** [review]

**Review by legleux — APPROVED** [review]

**Comment by intelliot** [comment]
@ckeshava confirmed this is ready from his perspective; adding `Passed` label.

**Comment by intelliot** [comment]
@ckeshava @legleux - did you confirm that this PR, as-is, fully fixes the issue reported in https://github.com/XRPLF/rippled/issues/4557#issue-1742480269?

Specifically:
> 1. All ports removed and isolated down to grpc_port - node stop
> 2. Isolated down to grpc_port and add protocol - node stop
> 3. grpc_port removed from server block - successful boot


In all of those cases, with this PR, the node will successfully boot with a correctly configured grpc_port, right?

**Comment by intelliot** [comment]
Suggested commit message:
```
feat: allow `[port_grpc]` to be specified in `[server]` stanza (#4728)

Instead of validating gRPC IP/Port/Protocol information in
ServerHandler, validate grpc port info in GRPCServer constructor. This
should not break backwards compatibility.

gRPC-related config info must be in a section (stanza) called
[port_gprc]. Prior to this commit, this section could not be added to
the [server] stanza.

* Close #4015 - That was an alternate solution. It was decided that with
  relaxed validation, it is not necessary to rename port_grpc.
* Fix #4557
```

**Comment by legleux** [comment]
:+1: 
With this PR, `rippled` no longer stops with 
```
Application:ERR Missing 'protocol' in [port_grpc]
```
when `port_grpc` appears in the `[server]` block which was the cause of confusion.

**Comment by intelliot** [comment]
Thanks. I think that's ok - in fact it's probably what we want, as the change should be 100% backward-compatible with any existing working cfg.

**Comment by intelliot** [comment]
Suggested commit message (updated):
```
feat: allow `port_grpc` to be specified in `[server]` stanza (#4728)

Prior to this commit, `port_grpc` could not be added to the [server]
stanza. Instead of validating gRPC IP/Port/Protocol information in
ServerHandler, validate grpc port info in GRPCServer constructor. This
should not break backwards compatibility.

gRPC-related config info must be in a section (stanza) called
[port_gprc].

* Close #4015 - That was an alternate solution. It was decided that with
  relaxed validation, it is not necessary to rename port_grpc.
* Fix #4557
```

**Comment by ckeshava** [comment]
@legleux  Thanks for verifying the output of this PR! I was not able to look into this one for the past few days

**Pull request merged**